### ISSUE TITLE
rocm packages do not seem to be available?

### ISSUE THREAD

**Issue by anoother** [nixos-rocm/nixos-rocm#40]
Hi,

I have followed the instructions in the README, but do not seem to be able to use any ROCM tools:

```
[root@minilith:~]# nix-shell -p rocminfo --run rocminfo
error: undefined variable 'rocminfo' at (string):1:94
(use '--show-trace' to show detailed location information)

[root@minilith:~]# nix-shell -p rocm-opencl-runtime --run clinfo
error: undefined variable 'rocm-opencl-runtime' at (string):1:94
(use '--show-trace' to show detailed location information)
```

```
[root@minilith:~]# rocm-smi
rocm-smi: command not found
```

My configuration for ROCM is here: https://github.com/anoother/etc-nixos/blob/master/rocm.nix This file is imported via `minilith.nix`, which is in turn imported by `configuration.nix`. The `nixos-rocm` repo is placed beside these files in `/etc/nixos` as a git submodule.

I certainly saw ROCM being built during `nixos-rebuild`, and have rebooted before trying the above commands. My kernel version is `5.4.43`.

I'm sure I must be missing something simple, so apologies for the distraction!

**Comment by acowley** [comment]
If it built during `nixos-rebuild`, then I think the overlay is not getting picked up for your user account.

**Comment by anoother** [comment]
Ah, I see. I will have a read of the docs; thought overlays in `configuration.nix` would be system-wide.

**Comment by acowley** [comment]
It's really confusing, and I can easily say something wrong about this so take anything I say with a healthy grain of salt. I have overlays listed in `configuration.nix`, but I also have an `overlays.nix` in `${HOME}/.config/nixpkgs/overlays.nix`.

Something useful about that is that my `overlays.nix` is something I use on non-NixOS systems, too.

I think an important phrase from the manual is,

> On a NixOS system the value of the nixpkgs.overlays option, if present, is passed to the system Nixpkgs directly as an argument. Note that this does not affect the overlays for non-NixOS operations (e.g. nix-env), which are looked up independently.

If `nix-shell` is a non-NixOS operation, then that could be what's happening to you.

**Comment by pl-semiotics** [comment]
Yes, you will need the overlay in `~/.config/nixpkgs/` for it to be picked up by user nix commands. I use the `~/.config/nixpkgs/overlays/` directory, which I link all the directories in `/etc/nixos` to.

When Nixpkgs is imported it only loads the options specified in `~/.config/nixpkgs`. Config options from `configuration.nix` are only parsed when one loads nixpkgs by means of nixos. Notice that without the entries in your home directory, `nix shell '(import <nixpkgs> {}).rocminfo"` does not work, while `nix shell '(import <nixpkgs/nixos> {}).pkgs.rocminfo'` does.

**Comment by Wulfsta** [comment]
I [hit this](https://github.com/nixos-rocm/nixos-rocm/issues/13#issuecomment-474145590) at one point as well, and also solved it by symlinking `/etc/nixos/overlays` to `~/.config/nixpkgs/overlays`. I think that's the laziest way to go about this, and is close to what you expected to see with a system wide overlay directory.

**Comment by DieGoldeneEnte** [comment]
There is also this trick from the [Nixos wiki](https://nixos.wiki/wiki/Overlays#Using_nixpkgs.overlays_from_configuration.nix_as_.3Cnixpkgs-overlays.3E_in_your_NIX_PATH)

It allows you to use the overlays from configuration.nix as user in nix-shell (nix search doesn't pick them up).
